samples/mongo-sample/tye.yaml

```diff
@@ -1,3 +1,4 @@
+name: mongo-sample
```

Comment:
@jkotalik should this be required?

Reply:
Right now the schema says its required, and deployment relies on it. If we made it optional, what would the defualt be? the directory name converted to lower?

Reply:
I just want to make sure it fails if it's not there.

Reply:
Ok so the answer here is that `ApplicationFactory` will default to the directory name. So it's not required. I need to update the docs and schema.

Do you see a drawback to it being optional? I'm cool with it.

Reply:
I'm fine with it being optional and using the directory name as a default.